### PR TITLE
Start adding LibraryReader.prefixForType.

### PR DESCRIPTION
**Pull request by matanlurey**
Work towards https://github.com/dart-lang/source_gen/issues/292.

I've made the feature package-private so we can iterate. It's possible we won't even be able to launch this as-is (if `computeNode()` explodes), but at least we will have the test case(s) when we have a replacement API, so I think it's worth checking (most) of this in.

### CODE REVIEW

File: lib/src/library.dart [hunk 1]
```diff
@@ -74,18 +74,60 @@ class LibraryReader {
...
+  /// Returns the identifier prefix of [element] if it is referenced by one.
+  ///
+  /// For example in the following file:
+  /// ```
+  /// import 'bar.dart' as bar;
+  ///
+  /// const b = const bar.Bar();
+  /// ```
+  ///
+  /// ... we'd assume that `b`'s element has a prefix of `bar`.
```
Comment (natebosch) [inline lib/src/library.dart]: I'm still confused by this. There isn't anywhere we'd have the code `bar.b` - so why is it that `b` is considered to have a prefix?

File: test/library_reader_test.dart [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  group('top level fields', () {
+    List<Element> topLevelFieldTypes;
+
+    setUpAll(() {
+      topLevelFieldTypes = b.element.definingCompilationUnit.topLevelVariables;
+    });
+
+    test('should read the prefix of a type', () {
+      expect(testingPrefixForType(b, topLevelFieldTypes[0]), 'a_prefixed');
```
Comment (natebosch) [inline test/library_reader_test.dart]: Can we look for specific variables rather than rely on the ordering?

File: lib/src/library.dart (hunk 1 above)
Comment (matanlurey) [reply lib/src/library.dart]: Yeah this isn't an accurate example anymore. I'll update it.

File: test/library_reader_test.dart (hunk 2 above)
Comment (matanlurey) [reply test/library_reader_test.dart]: Sure.

Comment (matanlurey) [reply test/library_reader_test.dart]: Done.

File: lib/src/library.dart (hunk 1 above)
Comment (matanlurey) [reply lib/src/library.dart]: Done.

**Pull request merged**